app/gui/src/controller/ide/desktop.rs

```diff
@@ -142,8 +157,9 @@ impl ManagingProjectAPI for Handle {
         async move {
             let logger = &self.logger;
             let project_mgr = self.project_manager.clone_ref();
-            let new_project = model::project::Synchronized::new_opened(logger, project_mgr, id);
-            self.current_project.set(new_project.await?);
+            let new_project =
+                model::project::Synchronized::new_opened(logger, project_mgr, id).await?;
```

Comment:
Are you able to refactor this line in such a way that it is not broken into 2 lines? You can create some additional variables if needed. Our code style tells about a strong preference towards single line expressions over multi-line ones.